port/zephyr/Kconfig

```diff
@@ -11,7 +11,7 @@ menuconfig GOLIOTH_FIRMWARE_SDK
 	select REBOOT if BOOTLOADER_MCUBOOT
 	select REQUIRES_FULL_LIBC # libcoap
 	select MBEDTLS
-    select MBEDTLS_SSL_DTLS_CONNECTION_ID
+    select MBEDTLS_SSL_DTLS_CONNECTION_ID if MBEDTLS_BUILTIN
```

Comment:
I think in practice `MBEDTLS_BUILTIN` is not set when we are using `nrf-sdk`, but it _could_ be. In that case I believe we could encounter issues -- are we okay with that for the time-being?

Reply:
Yeah, I'm okay with it. I think the chances of any of our current customers mucking with these settings is pretty low, and there's probably other things that would break too. We can revisit later, but I'm fine saying "using `MBEDTLS_BUILTIN` with NCS is unsupported" for now.